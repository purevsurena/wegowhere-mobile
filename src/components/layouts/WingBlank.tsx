import React from 'react';
import { View } from 'react-native';
import { WingBlankProps } from '@/types/layouts';

function WingBlank(props: WingBlankProps): JSX.Element {
  const { size = '16', style = {}, children, testID } = props;
  return (
    <View
      testID={testID}
      style={{
        marginLeft: parseInt(size, 10),
        marginRight: parseInt(size, 10),
        ...style,
      }}>
      {children}
    </View>
  );
}

export default WingBlank;
