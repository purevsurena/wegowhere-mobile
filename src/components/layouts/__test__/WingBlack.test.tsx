import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';
import WingBlank from '@/components/layouts/WingBlank';
import { Text } from 'react-native';

describe('<WingBlank />', () => {
  it('renders correctly with children', () => {
    const tree = renderer
      .create(
        <WingBlank size="20">
          <Text>Child</Text>
        </WingBlank>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children', () => {
    const { getByText } = render(
      <WingBlank size="16">
        <Text>Test Children</Text>
      </WingBlank>,
    );
    expect(getByText('Test Children')).toBeTruthy();
  });
});
