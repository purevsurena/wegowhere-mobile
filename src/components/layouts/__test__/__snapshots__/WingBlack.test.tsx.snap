// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WingBlank /> renders correctly with children 1`] = `
<View
  style={
    {
      "marginLeft": 20,
      "marginRight": 20,
    }
  }
>
  <Text>
    Child
  </Text>
</View>
`;
